lib/manifest.js

```diff
@@ -0,0 +1,78 @@
+let path = require("path");
+let { createFile, generateFingerprint, uriJoin } = require("faucet-pipeline-util");
+let retry = require("./retry");
+let retryTimings = [10, 50, 100, 250, 500, 1000];
+
+class Manifest {
+	constructor(config, fingerprint, configDir, exitOnError) {
+		this.config = config;
+		this.manifest = {};
+		this.fingerprint = fingerprint;
+		this.configDir = configDir;
+		this.exitOnError = exitOnError;
+	}
+
+	add(targetPath, content, opts = {}) {
+		let error = opts.error;
+		let outputPath = targetPath;
+
+		if(this.fingerprint) {
+			let fileName = path.basename(targetPath);
+			outputPath = path.join(path.dirname(targetPath),
+					generateFingerprint(fileName, content));
+		}
+
+		return createFile(outputPath, content).then(_ => {
+			this.report(outputPath, error);
+			return this.writeManifest(targetPath, outputPath);
+		}).then(_ => {
+			if(error && this.exitOnError) {
+				process.exit(1);
+			}
+		});
+	}
+
+	lookup(targetPath) {
+		return retry(this._lookup.bind(this), retryTimings)(targetPath);
+	}
+
+	_lookup(targetPath) {
+		return new Promise((resolve, reject) => {
+			if(this.manifest[targetPath]) {
+				resolve(this.manifest[targetPath]);
```

Comment:
IMO `this.manifest[targetPath]` should be cached in a variable _for readability_.